### PR TITLE
Change log level for cagg materialization messages

### PR DESCRIPTION
**Pull request by svenklemm**
The log level used for continuous aggregate materialization messages
was INFO which is for requested information. Since is no way to
control the behaviour externally INFO is a suboptimal choice because
INFO messages cannot be easily suppressed leading to irreproducable
test output. Even though time can be mocked to make output consistent
this is only available in debug builds.

This patch changes the log level of those messages to LOG, so
clients can easily control the ouput by setting client_min_messages.

### CODE REVIEW

**Comment by svenklemm** [comment]
> Looks OK, but I have some questions/comments:
> 
> 1. Most of these log messages seem almost like debug messages. Most people (including myself) I bet can't make a lot of sense of these messages because they are too much about internal details. Maybe they should be, e.g., `DEBUG1` rather than `NOTICE`? `NOTICE` seems too verbose, even if it can be turned off. It would also turn off other things that might be important.
> 2. The messages are covered by a `verbose` flag. Why can't that be used to suppress the output in tests?
> 
> Apart from this, there are a couple of typos in the commit message.

I talked about the log level with @gayyappan and she felt strongly that those are useful and should be shown by default. I would favor DEBUG1 here too but I dont mind the log level as long as it is not INFO.

While there is a verbose flag in our code this is not exposed. While some callers for the code could be adapted to expose the flag not all can.

**Comment by erimatnor** [comment]
> 
> I talked about the log level with @gayyappan and she felt strongly that those are useful and should be shown by default. I would favor DEBUG1 here too but I dont mind the log level as long as it is not INFO.
> 
> While there is a verbose flag in our code this is not exposed. While some callers for the code could be adapted to expose the flag not all can.

Actually, on further consideration I think LOG is appropriate. This would be emitted to server logs and if we need it sent to clients in this test we can set `client_min_messages=LOG`, if I am not mistaken. Wouldn't that also satisfy @gayyappan's concern? (Users could just check their server logs for these messages if we need to debug something for them).

From a UX perspective, emitting this to clients by default seems inappropriate.

**Comment by svenklemm** [comment]
If you wanna refactor the error messages i'm fine with that but this is not the PR for that.

**Comment by mkindahl** [comment]
> If you wanna refactor the error messages i'm fine with that but this is not the PR for that.

The key issue is that I think you should use the `ereport` for the message where you have a timestamp and put the timestamp in the detail. Otherwise, we cannot write tests that check that the message is printed when it should because the timestamp will vary all the time.

File: tsl/test/expected/continuous_aggs_materialize.out [hunk 1]
```diff
@@ -1055,21 +1054,17 @@ AS SELECT time_bucket('5', time), COUNT(data) as value
...
 ---------------------+-------
  9223372036854775800 |     1
 (1 row)
 
--- even when the subrtraction would make a completion time greater than max
+-- even when the subtraction would make a completion time greater than max
 INSERT INTO continuous_agg_negative VALUES (:big_int_max-1, 201);
 SET client_min_messages TO error;
 REFRESH MATERIALIZED VIEW negative_view_5;
-INFO:  new materialization range not found for public.continuous_agg_negative (time column time): not enough new data past completion threshold of 9223372036854775805 as of 9223372036854775805
```
Comment (gayyappan) [inline tsl/test/expected/continuous_aggs_materialize.out]: missing messages here.

File: tsl/test/isolation/expected/continuous_aggs_insert.out [hunk 2]
```diff
@@ -55,13 +48,11 @@ count
 step Refresh: REFRESH MATERIALIZED VIEW continuous_view; <waiting ...>
 step Sc: COMMIT;
 step UnlockCompleted: ROLLBACK;
-INFO:  materializing continuous aggregate public.continuous_view: nothing to invalidate, new range up to 15
```
Comment (gayyappan) [inline tsl/test/isolation/expected/continuous_aggs_insert.out]: I think we want to keep the messages in the isolation test as well.

Comment (JLockerman) [reply tsl/test/isolation/expected/continuous_aggs_insert.out]: To expand on this, part of what these tests check is when materialization can see concurrent INSERTs, and these messages are the only way we have to determine if the INSERTs were seen. (We use integer times for this, so the test _should_ be stable...)

**Pull request merged**